sdk/python/kfp/dsl/_pipeline_param.py

```diff
@@ -12,14 +12,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import re
-from collections import namedtuple
-from typing import List, Dict, Union
-
-# TODO: Move this to a separate class
-# For now, this identifies a condition with only "==" operator supported.
-ConditionOperator = namedtuple('ConditionOperator',
-                               'operator operand1 operand2')
-PipelineParamTuple = namedtuple('PipelineParamTuple', 'name op pattern')
+import attr
+from typing import List, Dict, Union, Any
+import warnings
+
+@attr.s
+class ConditionOperator:
+  operator = attr.ib(type=str)
+  operand1 = attr.ib(type=Union['PipelineParam', Any])
+  operand2 = attr.ib(type=Union['PipelineParam', Any])
```

Comment:
Can we use normal `__init__` or typed class variables?
What is the advantage of using attrs over the standard Python typing?

Reply:
>What is the advantage of using attrs

It provides `__str__`, `__repr__`, `__eq__`, `__hash__` and lots of other utilities out of the box.


> Can we use normal `__init__`

Normal `__init__` will provide neither of the good stuff mentioned above.


> or typed class variables?

We don't use any class variables here, these are all instance variables. Class variables are introduced via `ClassVar[T]`.